test/odbc/toby_tinyint_pg_datatype.cpp

```diff
@@ -0,0 +1,585 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+#include <iostream>
+using std::pair;
+
+const string TABLE_NAME = "master_dbo.tinyint_table_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.tinyint";
+const string VIEW_NAME = "master_dbo.tinyint_view_odbc_test";
+
+
+vector<pair<string, string>> TABLE_COLUMNS_TINYINT = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+class PSQL_DataTypes_int : public testing::Test{
+
+  void SetUp() override {
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+
+  void TearDown() override {
+
+    OdbcHandler test_teardown;
+    test_teardown.ConnectAndExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+    test_teardown.CloseStmt();
+    test_teardown.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// helper function to convert string to equivalent C version of int (long int)
+long int StringToTinyInt(const string &value) {
+  return strtol(value.c_str(), NULL, 10);
+}
+
+TEST_F(PSQL_DataTypes_int, Table_Creation) {
+
+  const int LENGTH_EXPECTED = 3;
+  const int PRECISION_EXPECTED = 0;
+  const int SCALE_EXPECTED = 0;
+  const string NAME_EXPECTED = "NULL";
+  const int BYTES_EXPECTED = 1;
+
+  const int BUFFER_SIZE = 256;
+  char name[BUFFER_SIZE];
+  SQLLEN length;
+  SQLLEN precision;
+  SQLLEN scale;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_TINYINT));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure column attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_DISPLAY_SIZE, // Get the length of the column (size of char in columns)
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  // ASSERT_EQ(length, LENGTH_EXPECTED);    refer to babelfish_extensions/contrib/babelfishpg_common/sql/numerics.sql
```

Comment:
Given that we are testing if the `sys` datatype (i.e. `sys.tinyint`) conforms to the underlying PG type (i.e. `smallint`) we expect the length for `sys.tinyint` to match that of PG type's `smallint`. So I think we can uncomment out this assert.